inception-doc/src/main/resources/META-INF/asciidoc/developer-guide/release.adoc

```diff
@@ -0,0 +1,101 @@
+= Releasing INCEpTION
+
+== Prerequisites
+
+When deploying to an Artifactory instance, make sure that the UKP Maven Repository is with your
+user credentials are added to your machines `settings.xml`. In our case, the repository
+`ukp-oss-releases` has to be listed with a user and password that belongs to the `trusted-user`
+group. See https://dkpro.github.io/dkpro-core/pages/setup-maven[DKPro Core - Setting up Maven repository]
+for more information on how to do this.
+
+== Release steps
+
+[arabic]
+. Create a release issue in the INCEpTION Github issue tracker
+. Merge the branch you want to release (e.g. `{revnumber}`) into the
```

Comment:
This is of course only necessary if the branch you want to release is not master? Normally we would not release master though. So a prior step needs to be added about creating a new maintenance branch if a major feature release is being made - not relevant if a bugfix release is being made.

The guide should explain the versioning scheme and bugfix and feature releases are.

Note that this is only possible if the release manager has write permissions to master - otherwise this must be done via a PR which must be reviewed. If the release manager has write permissions to master, then doing a PR for the release is not really necessary because it can be done directly on master.